### PR TITLE
Opt: meowfficer refactor

### PR DESCRIPTION
**Pull request by nEEtdo0d**
- break into several files and inherit at last level
- enhance to clear up cattery space
- queue alternative to use up common and elite boxes first
- sr cat check; lock if has unique or rainbow skill else skip and allow it to be used as feed material

Did a isolated testing across the meowfficer functions and seems stable.  I will close the previous PR.

### CODE REVIEW

File: module/meowfficer/enhance.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        """
+        # Base Cases
+        # - Config at least > 0
+        # - Coins at least > 1000
+        if self.config.Meowfficer_EnhanceIndex <= 0:
+            return
+
+        coins = MEOWFFICER_COINS.ocr(self.device.image)
+        if coins < 1000:
+            return
```
Comment (LmeSzinc) [inline module/meowfficer/enhance.py:180]: Add `logger.hr('Meowfficer enhance', level=1)` at the beginning of `meow_enhance()`.
Log each if case explaining why to quit.
`logger.warning('Invalid Meowfficer_EnhanceIndex={self.config.Meowfficer_EnhanceIndex}, skip')`
`logger.info('Coins ({coins}) < 1000, may not have enough coins to enhance, skip')`

File: module/meowfficer/enhance.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+                      additional=self.meow_additional, retry_wait=3, skip_first_screenshot=True)
+        self.ui_click(MEOWFFICER_FEED_ENTER, check_button=MEOWFFICER_FEED_CONFIRM,
+                      additional=self.meow_additional, retry_wait=3, skip_first_screenshot=True)
+
+        # Initiate feed sequence
+        # - Select Feed
+        # - Confirm/Cancel Feed
+        # - Confirm Enhancement
+        self.meow_feed_select()
+        self.meow_enhance_confirm()
```
Comment (LmeSzinc) [inline module/meowfficer/enhance.py:199]: Need a while loop to enhance until meowfficers are exhausted.

Also add `logger.hr('Eenhance once', level=2)` for each loop, add `logger.info('Meowfficer feed select')` at the beginning of `meow_feed_select()`, add `logger.info('Meowfficer enhance confirm')` at the beginning of `meow_enhance_confirm()`.
The log should end up like this:
```
===== Meowfficer enhance =====
meow_select
...
----- Enhance once -----
Meowfficer feed select
...
Meowfficer enhance confirm
...
----- Enhance once -----
Meowfficer feed select
...
No more meowfficers to select
```

**Review by LmeSzinc — COMMENTED** [review]
An apperance improvement:
Add logs to show what it's doing and why it's entring an IF branch. I gave an example on the enhance part and the rest parts should be done like that.

After that, I'll merge into dev branch first and update assets later.

**Comment by nEEtdo0d** [comment]
Sorry for the delay, busy week. But I think I covered all your comments and some other optimizations, stable after some testing.  I've redirected the PR to dev branch, a bit out of date it seems but this PR would introduce afterwards.

File: module/meowfficer/collect.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+                return True
+
+            if self.appear(MEOWFFICER_TRAIN_START, offset=(20, 20)):
+                return True
+
+            return False
+
+        self.ui_click(MEOWFFICER_CONFIRM if has_unique else MEOWFFICER_CANCEL,
+                      check_button=check_popup_exit, offset=(40, 20),
+                      retry_wait=3, skip_first_screenshot=True)
```
Comment (LmeSzinc) [inline module/meowfficer/collect.py:108]: I guess these are server specific, but would still like to confirm. CN has popups on SSR meowfficers only while SR don't. Do EN has popups on both SR and SSR meowfficers?

**Pull request merged**